client/src/bc/landing_page.gleam

```diff
@@ -1,23 +1,51 @@
+import gleam/dynamic
+import gleam/io.{debug}
+import gleam/option.{type Option, None, Some}
 import lustre
 import lustre/attribute.{class, src}
-import lustre/element.{text}
-import lustre/element/html.{a, div, h1, img, li, nav, p, ul}
+import lustre/effect.{type Effect}
+import lustre/element.{type Element, text}
+import lustre/element/html.{a, button, div, h1, img, li, nav, p, ul}
+import lustre/event
+import lustre_http.{type HttpError}
 import sketch
 import sketch/options as sketch_options
 
-pub fn app() {
-  let assert Ok(render) =
-    sketch_options.node()
-    |> sketch.lustre_setup()
-  lustre.simple(init, update, render(view))
+// Model
+pub type Model {
+  Model(ping: Option(Ping))
+}
+
+pub type Ping {
+  Ping(ping: String)
+}
+
+pub fn init(_) -> #(Model, Effect(Msg)) {
+  #(Model(ping: None), effect.none())
+}
+
+// Update
+
+pub opaque type Msg {
+  UserClickedPing
+  ApiUpdatedPing(Result(Ping, HttpError))
 }
 
-pub fn init(_) {
-  0
+pub fn update(model: Model, msg: Msg) -> #(Model, Effect(Msg)) {
+  debug(msg)
+  case msg {
+    UserClickedPing -> #(model, ping())
+    ApiUpdatedPing(Ok(ping)) -> #(Model(ping: Some(ping)), effect.none())
+    ApiUpdatedPing(Error(_)) -> #(model, effect.none())
+  }
 }
 
-pub fn update(model, _msg) {
-  model
+fn ping() -> Effect(Msg) {
+  // fixme: use the real url here
+  let url = "http://localhost:8000/ping"
```

Comment:
This shouldn't be in the deployed application. Any suggestions on what we could do for the testing vs actual prod url should be? Maybe a command line argument?

Reply:
Usually, you use an env variable, but in the case of a pre-built set of javascript served to the browser, there aren't any env variables, so you usually default to the schema/domain/port that the app was served on.  In general, it's actually best if this is the way it's also served locally as well

Reply:
I don't know how to get access to the current scheme/domain/port in gleam, but you can certainly get access to it in Javascript, so it should be possible.

In Elm, I think the recommended way to do things like this was to have the javascript that does the initial page load generate this kind of thing and then pass it in on init.

Reply:
Alright I think I came up with a solution for this. Let me know what you think